Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift

```diff
@@ -0,0 +1,308 @@
+        galleryButton.titleLabel?.font = qrCodeReaderDataModel.galleryButtonModel.font
+        galleryButton.isHidden = qrCodeReaderDataModel.galleryButtonModel.isHidden
+        galleryButton.addTarget(self, action: #selector(galleryButtonTapped), for: .touchUpInside)
+    }
+    
+    private func configureQRCodeReader() {
+        captureSession = AVCaptureSession()
+        guard let videoCaptureDevice = AVCaptureDevice.default(for: .video) else { return }
+        let videoInput: AVCaptureDeviceInput
+        
+        do {
+            videoInput = try AVCaptureDeviceInput(device: videoCaptureDevice)
+        } catch {
+            return
+        }
+        
+        if captureSession.canAddInput(videoInput) {
+            captureSession.addInput(videoInput)
+        } else {
+            captureSession = nil
+            delegate?.qrCodeReaderFailed(self)
+            return
+        }
+        
+        if captureSession.canAddOutput(metadataOutput) {
+            captureSession.addOutput(metadataOutput)
+            metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+            metadataOutput.metadataObjectTypes = [.qr]
+        } else {
+            captureSession = nil
+            delegate?.qrCodeReaderFailed(self)
+            return
+        }
+        
+        previewLayer = QRCodeReaderPreviewLayer(session: captureSession,
+                                                marginSize: qrCodeReaderDataModel.qrCodeReaderPreviewLayerModel.marginSize,
+                                                lineWidth: qrCodeReaderDataModel.qrCodeReaderPreviewLayerModel.lineWidth,
+                                                lineColor: qrCodeReaderDataModel.qrCodeReaderPreviewLayerModel.lineColor,
+                                                lineDashPattern: qrCodeReaderDataModel.qrCodeReaderPreviewLayerModel.lineDashPattern,
+                                                cornerRadiusSize: qrCodeReaderDataModel.qrCodeReaderPreviewLayerModel.cornerRadius)
+        previewLayer.frame = view.layer.bounds
+        previewLayer.backgroundColor = qrCodeReaderDataModel.qrCodeReaderPreviewLayerModel.backgroundColor
+        previewLayer.videoGravity = .resizeAspectFill
+        view.layer.addSublayer(previewLayer)
+        
+        view.bringSubviewToFront(closeButton)
+        view.bringSubviewToFront(infoLabel)
+        view.bringSubviewToFront(galleryButton)
+        
+        /// Captures only inside of reader square
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.view.layoutIfNeeded()
+            self.view.setNeedsLayout()
+            self.metadataOutput.rectOfInterest = self.previewLayer.rectOfInterest
+        }
+    }
+}
+
+// MARK: - Actions
+extension QRCodeReaderViewController {
+    
+    @objc
+    private func closeButtonTapped() {
+        dismiss(animated: true)
```

Comment:
Can be added **qrCodeReaderDismiss** func to **QRCodeReaderDelegate** and call from here.